### DISCUSSION TITLE
differnt acme challanges?

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#2827] [Questions] [answered]
https://github.com/nextcloud/all-in-one/blob/b7c5e0b95314ee67631d62b70b9fa794877dcc2c/Containers/apache/Caddyfile#L65
https://github.com/nextcloud/all-in-one/blob/b7c5e0b95314ee67631d62b70b9fa794877dcc2c/Containers/mastercontainer/Caddyfile#L31
I don't understand the sense of this, why there are different challenges selected, would it not be better to use the same?

**Comment by szaimen** [comment, marked as answer]
> https://github.com/nextcloud/all-in-one/blob/b7c5e0b95314ee67631d62b70b9fa794877dcc2c/Containers/apache/Caddyfile#L65

The apache container uses the tlsalpn_challenge  which is only possible via port 443

> https://github.com/nextcloud/all-in-one/blob/b7c5e0b95314ee67631d62b70b9fa794877dcc2c/Containers/mastercontainer/Caddyfile#L31

The mastercontainer uses the http_challenge which is only possible via port 80

> I don't understand the sense of this, why there are different challenges selected, would it not be better to use the same?

The main idea is that mastercontainer and apache are completely separate from each other. So e.g. if cert-renewal for apache fails, it should usually not effect the AIO interface.
